crux/src/Crux/UI/JS.hs

```diff
@@ -3,33 +3,61 @@
 -- | Utilites for generating JSON
 module Crux.UI.JS where
 
-import Data.Text(unpack)
+import Data.Text(unpack, Text)
+import qualified Data.Text as Text
+import Numeric
 import Data.List(intercalate)
 import Data.Maybe(fromMaybe)
 import System.Directory( canonicalizePath )
 
+import Data.Aeson.Encode.Pretty (encodePretty)
+import Data.Aeson (decode, Value)
+import qualified Data.ByteString.Lazy.Char8 as BL
+
+
 import What4.ProgramLoc
 
+-- |`jsLoc` takes a program location, and optionally returns it
```

Comment:
"Optionally returns" is a strange way to describe this, as this is a function that can potentially fail (by returning `Nothing`). I would phrase this as:

> `jsLoc` takes a program location and renders it as a JavaScript string. This returns `Nothing` if it is unclear how to render the program location.

Reply:
Done